src/ui/public/test_harness/test_sharding/get_shard_num.js

```diff
@@ -0,0 +1,28 @@
+import murmurHash3 from 'murmurhash3js';
+
+// murmur hashes are 32bit unsigned integers
+const MAX_HASH = Math.pow(2, 32);
+
+/**
+*  Determine the shard number for a suite by hashing
+*  it's name and placing it based on the hash
+*
+*  @param {number} shardTotal - the total number of shards
+*  @param {string} suiteName - the suite name to hash
+*  @return {number} shardNum - 1-based shard number
+*/
+export function getShardNum(shardTotal, suiteName) {
```

Comment:
It's *used* in karma.js, but both the concept of `shard number` and the variable name `shard_num` are defined in and fundamental to this `test_sharding` module.  They're not different at all, the value this represents is exactly the same concept as the value shard_num represents.